### PR TITLE
Add a Galleon feature pack to allow Galleon provisioning of the SAML adapter to WildFly

### LINKED ISSUE #12363: Provide a Galleon feature pack to install the Keycloak Elytron SAML adapter
### Description

A new Gallon feature pack should be added that contains a single layer, `keycloak-client-saml`.

This relates to [WFLY-16306](https://issues.redhat.com/browse/WFLY-16306).

I will work on this.

### Discussion

_No response_

### Motivation

This will make it easier to secure applications deployed to WildFly with SAML without needing to manually install the Keycloak Elytron SAML adapter using the zip file. 

### Details

A similar feature pack to install the Keycloak Elytron OIDC adapter was added a little while ago before WildFly
added native support for OpenID Connect (see [WFLY-13706](https://issues.jboss.org/browse/WFLY-13706) / [KEYCLOAK-14953](https://issues.jboss.org/browse/KEYCLOAK-14953)).

For more details, see https://github.com/wildfly/wildfly-proposals/pull/486.

### PR DESCRIPTION
**Pull request by fjuma**
This PR adds a Galleon feature pack to allow Galleon provisioning of the SAML adapter to WildFly. More details about this can be found [here](https://github.com/wildfly/wildfly-proposals/pull/486).

This PR also includes @pskopek's changes for transitioning the SAML adapter to Jakarta EE 10 since that's a prerequisite for this work.

Closes https://github.com/keycloak/keycloak/issues/12363

### CODE REVIEW

**Comment by fjuma** [comment]
@pskopek Just noticed the conflict in the pom.xml file. I based this branch off your `saml-adapter-jakarta-transition` branch since those changes aren't in `main` yet. If I rebase to resolve the conflict, the commit ID for your commit that's also included in this PR will change.

**Comment by fjuma** [comment]
@jfdenise Just FYI, my changes for adding a Galleon feature pack to allow Galleon provisioning of the SAML adapter to WildFly are now complete. Please review when you get a chance, thanks!

File: distribution/galleon-feature-packs/saml-adapter-galleon-pack/wildfly-feature-pack-build.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        <dependency group-id="org.wildfly" artifact-id="wildfly-ee-galleon-pack">
+            <name>org.wildfly:wildfly-ee-galleon-pack</name>
+            <packages inherit="false">
+                <exclude name="product.conf"/>
+            </packages>
+            <default-configs inherit="false"/>
+        </dependency>
+    </transitive>
+    <dependencies>
+        <dependency group-id="org.wildfly" artifact-id="wildfly-galleon-pack">
```
Comment (jfdenise) [inline distribution/galleon-feature-packs/saml-adapter-galleon-pack/wildfly-feature-pack-build.xml:28]: @fjuma , do you have a need for a dependency on wildfly-galleon-pack? Could it depend only on wildfly-ee-galleon-pack?

Comment (fjuma) [reply distribution/galleon-feature-packs/saml-adapter-galleon-pack/wildfly-feature-pack-build.xml]: @jfdenise Thanks for pointing that out! I've now updated this to depend only on `wildfly-ee-galleon-pack`.

**Review by jfdenise — APPROVED** [review]

**Comment by tommaso-borgato** [comment]
@fjuma @jfdenise a question about Jakarta 10 : it looks to me that this MR is compatible with Jakarta EE 10 ... is that correct? in case it is, is there any plan to make it work with the older specification (Jakarta EE 8)?

**Comment by fjuma** [comment]
> @fjuma @jfdenise a question about Jakarta 10 : it looks to me that this MR is compatible with Jakarta EE 10 ... is that correct? in case it is, is there any plan to make it work with the older specification (Jakarta EE 8)?

@tommaso-borgato Correct. The commit that transitions the SAML adapter to Jakarta EE 10 has also been included in this PR and can be seen here:
https://github.com/keycloak/keycloak/pull/15964/commits/34a2f87aad76d1d0e93b680b149ba25e02998a24

No plans to make this work with the older specification.

**Review by stianst — CHANGES_REQUESTED** [review]
I'm slightly confused about the changes in some classes from javax to jakarta. One the one side this will break the adapters using these classes for those not switching to JakartaEE, and on the other hand it seems insufficient to those that will.

**Comment by fjuma** [comment]
@stianst The changes from javax to jakarta that are included in this PR are from @pskopek's [commit](https://github.com/keycloak/keycloak/pull/15964/commits/34a2f87aad76d1d0e93b680b149ba25e02998a24) that transitions the SAML adapter to EE 10. I needed to include those changes in this PR since these changes aren't in the `main` branch yet but are needed to work with the latest version of WildFly.

**Comment by pskopek** [comment]
@fjuma this PR can be closed. See https://github.com/keycloak/keycloak/pull/19710

**Comment by fjuma** [comment]
Thanks @pskopek, closing it.

**Pull request closed without merging**